src/modules/payment/entities/payment.entity.ts

```diff
@@ -0,0 +1,47 @@
+import { Booking } from '#/modules/bookings/entities/booking.entity';
+import { PaymentStatus } from '#/shared/constants/payment-status.constant';
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  JoinColumn,
+  OneToOne,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+
+@Entity('payment')
+export class Payment {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column({ nullable: false })
+  amount: number;
+
+  @Column({ type: 'enum', enum: PaymentStatus })
+  status: PaymentStatus;
+
+  @Column({ name: 'transaction_id' })
+  transactionId: string;
+
+  @CreateDateColumn({
+    name: 'created_at',
+    type: 'timestamp',
+    default: () => 'CURRENT_TIMESTAMP',
+  })
+  createdAt: Date;
+
+  @UpdateDateColumn({
+    name: 'updated_at',
+    type: 'timestamp',
+    default: () => 'CURRENT_TIMESTAMP',
+  })
+  updatedAt: Date;
+
+  @OneToOne(() => Booking, (booking) => booking.id, { onDelete: 'SET NULL' })
```

Comment:
```(booking) => booking.id``` this expression is wrong. We should write here by which field name we can get this payment relation when getting the booking. Correct expression would be ```(booking) => booking.payment```, but after that we need to also add ```payment``` field to ```booking``` table as well. Let's for now remove this, in the future we can add both.

Reply:
fixed